internal/engine/dcwatch/disable_watcher.go

```diff
@@ -77,29 +83,43 @@ func (w *DisableSubscriber) OnChange(ctx context.Context, st store.RStore, summa
 		isDisabled := ms.DisableState == v1alpha1.DisableStateDisabled
 		needsCleanup := isRunning && isDisabled
 
-		anyNeedCleanup = anyNeedCleanup || needsCleanup
-
 		rs := resourceState{
 			Spec:         mt.Manifest.DockerComposeTarget().Spec,
 			NeedsCleanup: needsCleanup,
 			StartTime:    ms.DCRuntimeState().StartTime,
 			Order:        i,
 		}
 		w.mu.Lock()
-		w.resourceStates[rs.Spec.Service] = rs
+		rs.CurrentlyCleaningUp = w.resourceStates[name].CurrentlyCleaningUp
+		if rs.NeedsCleanup && !rs.CurrentlyCleaningUp {
+			rs.CurrentlyCleaningUp = true
+			toCleanup = append(toCleanup, name)
+		}
+		w.resourceStates[name] = rs
 		w.mu.Unlock()
 	}
 
 	st.RUnlockState()
 
-	if anyNeedCleanup {
+	if len(toCleanup) > 0 {
 		go func() {
+			w.mu.Lock()
+			w.goroutinesSpawnedForTesting += 1
```

Comment:
you probably want this to be outside the goroutine? then you don't need the lock here